gui/wxpython/startup/guiutils.py

```diff
@@ -211,6 +212,44 @@ def create_mapset_interactively(guiparent, grassdb, location):
     return mapset
 
 
+def create_location_interactively(guiparent, grassdb):
```

Comment:
I think this function should return tuple (database, location, mapset) where database and location are what it returns now and mapset is `"PERMANENT"` by default and result of `create_mapset_interactively()` in case `gWizard.user_mapset` is `True`. This way, the issue of additional mapset creation is encapsulated in this function.

The documentation should say that the return value, i.e., the tuple (database, location, mapset), is the location user created and the mapset is a mapset user may want to switch to. At this point it will be PERMANENT, but theoretically it could be another mapset the user created. The emphasis in "may want to switch to" is on the "may", it depends on the caller what the context is. In datacatalog, we currently don't switch in this case, in startup we change selection (or whatever is the current state, I would have to double check, but the point is, it is up to do caller to decide).

Reply:
create_location_interactively now returns tuple (database, location, mapset) where database and location are what it returns now and mapset is "PERMANENT" by default or the result of create_mapset_interactively() in case gWizard.user_mapset is True.

Reply:
This also contains some info on what should be in the documentation. To your:

```
        # Returns database and location created by user
        # and a mapset user may want to switch to
```

The point of "user may want to switch to" is that the function can create only PERMANENT or it can create one or (in the future) more other mapsets. The caller does not know and this function does not tell. What this function reports back is the mapset the user may want to switch to if any switching is expected (not the case currently in data catalog, but it is the case in the startup window). This mapset can be PERMANENT or anything else.

Reply:
Ok, so if I get it right we do not have any mechanism for mapset creation within the location wizard so far.  So it is meant for future realization. 
Docstring added.

Reply:
Actually, if something, we would switch outside of this function, hence giving the mapset name as a hint to the caller of this function - a hint about what user may want to do next. There is no difference for the caller between PERMANENT and any other mapset; as I said, _this function reports back is the mapset the user may want to switch...This mapset can be PERMANENT or anything else._ When user does not ask for anything else, only PERMANENT is created, so the function returns PERMANENT by default as mapset, but if user asks for creation of another mapset, the function returns name of this other mapset. In other words, _the function can create only PERMANENT or it can create...other mapset_.

To back up little further, the function is creating a new location, so returning its name (and path) is sufficient behavior. However, we are also adding mapset as additional information since other mapsets besides PERMANENT could be created based on user input as part of the location creation process.

Here, we are trying to create general interface in between two parts of the code which won't change when we do small changes in the code or behavior here or there. Thus, it is important to define what return values are in general way. Database and location are something the caller will likely need to use since a new location was just created, but mapset is something which the caller is less likely to use. The function is saying "Hey, FYI, this might a mapset the user is interested in the most."